File: vulnerable1/basic.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

struct student
{
    char name[10];
    char grade[5];
};

int main(int argc, char *argv[])
{
    struct student stu;

    strcpy(stu.grade, "nil");

    gets(stu.name);

    printf("Hi %s! Your grade is %s.\n", stu.name, stu.grade);

    exit(0);
}

File: vulnerable1/basic.rs
fn main() {
    let mut stu = Student {
        name: String::new(),
        grade: String::from("nil"),
    };

    std::io::stdin().read_line(&mut stu.name).expect("Failed to read name.");

    println!("Hi {}! Your grade is {}.", stu.name, stu.grade);
}

struct Student {
    name: String,
    grade: String,
}
